testsuite_overview.md

```diff
@@ -0,0 +1,310 @@
+|------------------|--------|
+| Primitive types|int, long, float, double, bytes, string |
+| Complex types | array of maps, map of ints, record with props |
+| Complex type: Unions | union of complex types, union of primitives, union of same types |
+| Enums | With props and with alias|
+| Fixed type | |
+| Typerefs | for string, array, chained typeref, array, map, field, union|
+| Include | include, and include with include |
+| Default Fixup | to see if required fields are "fixed up" with defaults|
+| Optional fields | |
+
+### Wire Protocol Tests
+These are tests for building requests and decoding responses. 
+
+We test for well-formed requests by comparing the built http request
+with the expected http request in the requests/ or requests-v2/ folder. We compare url, method, headers, and body.
+We test that Rest.li can decode a Rest.li response from an http response by checking the decoded Rest.li response for
+the correct values. We compare the response's status and error message with the status and error message specified by 
+manifest.json. The body of the response is tested through manual assertions that check for the correct values.
+
+##### Basic Resource Method Tests for Requests/Responses
+
+| Rest.li Method | Collection | Simple | Association | Action Set|
+|--------|------|----------------|----------------|------------------|
+| get | x | x | x | |
+| batch-get | x |  | x | |
+| finder | x | | x | |
+| create | x | | | |
+| create with returned entity | x | | | |
+| batch-create | x | | | |
+| update | x | x | x | |
+| partial update| x | |o| |
+| batch-update| x |  | x | |
+| batch-partial-update| x| | o | |
+| delete | x | x | x | | 
+| batch-delete | x | | x | | 
+| action | o |  o | o | x | 
+
+"x" - test is included in test suite  
+"o" - test is not included but method should be supported by the resource  
+" " - test is not included and method should NOT be supported by resource
+
+##### Resource Key Tests
+| Key Feature | Rest.li Method used|
+|-------------|---------------|
+|Key with Union| get | 
+|Query Params| get | 
+|Complex Key| get, create, delete, update batch-create, batch-delete, batch-get, batch-update, partial-update |
+|Special Chars in ComplexKey strings | get, batch-get|
+
+##### Error tests
+| Error | Resource used | Rest.li Method | Details |
+|-----|-----|----|----|
+|404 | Collection | get | Send empty get request|
+|400 | Collection | update | Request has a missing required field | 
+|500 | Collection | create | Create request with id field |
+|Error Details | Collection | create | CreateResponse with Error Details| 
+|Batch Results with Errors | Collection | batch_update | Batch update with one good and two bad requests| 
+
+##### Misc. Tests
+| Feature | Resource tested| Method used|
+|---------|---------|------|
+| Typeref | Collection, Association | get |
+| Subresource | Collection, Association | get |
+| Projection | Collection, Association, ComplexKeyResource | get | 
+
+
+Development Status
```

Comment:
This section is not needed anymore, and I assume that they should already be covered in your above TestSuite Specification Coverage.